client/components/ProductOverview/ImageGallery.jsx

```diff
@@ -1,26 +1,39 @@
-import React, { useState } from 'react';
+import React from 'react';
 import { AiOutlineArrowLeft, AiOutlineArrowRight } from 'react-icons/ai';
 import PropTypes from 'prop-types';
 
-const ImageGallery = ({ currentStyle }) => {
-  const [currentImageIndex, setCurrentImageIndex] = useState(0);
-  const currentPhoto = currentStyle.photos[currentImageIndex].url;
+import ThumbnailView from './ThumbnailView';
+
+const ImageGallery = ({ currentStyle, imageIndex, updateImageIndex }) => {
+  if (currentStyle === null) {
+    return <div>No data available</div>;
+  }
+
   const { photos } = currentStyle;
   const { length } = photos;
+  if (imageIndex > length) {
+    updateImageIndex(0);
+  }
+  const currentPhoto = currentStyle.photos[imageIndex].url;
 
   const handleLeftArrow = () => {
```

Comment:
Couldn't we have all of the logic live in one handler with a flag of either left or right?

Reply:
refactored